force-app/main/service/AddressService.cls

```diff
@@ -151,8 +155,18 @@ public inherited sharing class AddressService {
         sobjDst.put(strFieldPrefix + 'PostalCode', addr.MailingPostalCode__c);
         sobjDst.put(strFieldPrefix + 'Latitude', addr.Geolocation__Latitude__s);
         sobjDst.put(strFieldPrefix + 'Longitude', addr.Geolocation__Longitude__s);
-        sobjDst.put(strFieldPrefix + 'State', addr.MailingState__c);
-        sobjDst.put(strFieldPrefix + 'Country', addr.MailingCountry__c);
+        String countryField = 'Country';
+        String stateField = 'State';
+        if (orgConfig.isStateCountryPicklistsEnabled()) {
+            if (addr.MailingCountry__c != null && addr.MailingCountry__c.length() <= 3) {
```

Comment:
This may have been what we were doing before, but we are already checking in SCP are enabled at this point, so we should just check if the value in the address field is a valid state/country code or not and set it if it's a valid code. which I believe would require a describe on the state/country picklists. It's heavier-handed performance wise but more reliable than checking length